games-board/xboard/xboard-4.9.1.ebuild

```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit autotools eutils fdo-mime gnome2-utils
+
+DESCRIPTION="GUI for gnuchess and for internet chess servers"
+HOMEPAGE="https://www.gnu.org/software/xboard/"
+SRC_URI="mirror://gnu/xboard/${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE="Xaw3d +default-font gtk nls zippy"
+RESTRICT="test" #124112
+
+RDEPEND="
+	dev-libs/glib:2
+	games-board/gnuchess
+	gnome-base/librsvg:2
+	virtual/libintl
+	x11-libs/cairo[X]
+	x11-libs/libXpm
+	default-font? (
+		media-fonts/font-adobe-100dpi[nls?]
+		media-fonts/font-misc-misc[nls?]
+	)
+	!gtk? (
+		x11-libs/libX11
+		x11-libs/libXt
+		x11-libs/libXmu
+		Xaw3d? ( x11-libs/libXaw3d )
+		!Xaw3d? ( x11-libs/libXaw )
+	)
+	gtk? ( x11-libs/gtk+:2 )"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	x11-base/xorg-proto
+	nls? ( sys-devel/gettext )"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-gnuchess-default.patch
```

Comment:
Can you explain what this patch does to me? It looks like there are more than one engine that works with this program, but to me this patch hardcodes gnuchess as 'the only choice'. Is it so?

Reply:
In fact by default (when you check in the source), xboard uses "fairymax" as chess engine. But this engine is not available on Gentoo. So the creator of this patch replace "fairymax" by "gnuchess"